singer/src/main/java/com/pinterest/singer/writer/s3/ObjectUploaderTask.java

```diff
@@ -0,0 +1,71 @@
+package com.pinterest.singer.writer.s3;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+
+import java.io.File;
+
+public class ObjectUploaderTask {
+    public static final int MAX_RETRIES = 5;
+    private static final Logger LOG = LoggerFactory.getLogger(S3Writer.class);
+    private final S3Client s3Client;
+    private final String bucket;
+    private final String keyPrefix;
+    private final int maxRetries;
+    private static final long INITIAL_BACKOFF = 1000; // Initial backoff in milliseconds
+    private static final long MAX_BACKOFF = 32000; // Maximum backoff in milliseconds
+
+    public ObjectUploaderTask(S3Client s3Client, String bucket, String keyPrefix, int maxRetries) {
+        this.s3Client = s3Client;
+        this.bucket = bucket;
+        this.keyPrefix = keyPrefix;
+        this.maxRetries = maxRetries;
+    }
+
+    public boolean upload(File file, String fileFormat) {
+        int attempts = 0;
+        boolean success = false;
+        long backoff = INITIAL_BACKOFF;
+
+        while (attempts < maxRetries && !success) {
```

Comment:
can we make sure that maxRetries is not 0? i.e. if the user passes in 0 to the config for maxRetries, it is possible that no attempt is made at all to upload. the user might assume that `maxRetries=0` means that it should not retry if the first attempt fails, but the logic here will prevent the first upload from being attempted at all.

Reply:
```
if (s3WriterConfig.getMaxRetries() > 0) {
      this.maxRetries = s3WriterConfig.getMaxRetries();
    } else {
      this.maxRetries = ObjectUploaderTask.MAX_RETRIES;
    }
```

Referencing this code again, I believe if the user puts 0 or anything less (negative), it will default to 5. Do you think this needs to be corrected in anyway? Thanks Jeff!